### PR TITLE
Consolidates GH action build to a single file

### PR DESCRIPTION
**Pull request by codefromthecrypt**
This consolidates the two build processes into one flow, containing all
previous functionality, but less build time.

This builds the server on ubuntu, os/x and windows. If ubuntu, the flow
will continue into building and testing all our Docker images.

Something new is that this skips out if the change is documentation-only.

### CODE REVIEW

File: .github/workflows/continuous-build.yml [hunk 1]
```diff
@@ -1,19 +1,33 @@
...
       - master
+    paths-ignore:
+      - '**/*.md'
   pull_request:
+    branches:
+      - master
+    paths-ignore:
+      - '**/*.md'
+
+# DEFAULT_OS is ubuntu-20.04 (focal) as that's the same as we use in Travis. This is find/replaced
```
Comment (anuraaga) [inline .github/workflows/continuous-build.yml]: Not sure, but are you able to use a single-element item in the matrix as a variable?

```
matrix:
  ubuntu-os:
  - ubuntu-20.04

...

if: matrix.os == ${{ matrix.ubuntu-os }}
```

**Review by anuraaga — APPROVED** [review]

File: .github/workflows/continuous-build.yml (hunk 1 above)
Comment (codefromthecrypt) [reply .github/workflows/continuous-build.yml]: will try it! right now GH aren't charging us per run so why not give it a shot :D

Comment (codefromthecrypt) [reply .github/workflows/continuous-build.yml]: I think the last commit works. but GH actions is being annoying. will try to kick it with close/reopen

**Comment by codefromthecrypt** [comment]
trying squash and force push as re-open had no effect on the stalled actions

**Comment by codefromthecrypt** [comment]
last fail was legit at least parsing wise as it complained that yaml anchors are unsupported

**Comment by codefromthecrypt** [comment]
ok I think this is as good as it gets. you can't use yaml anchors, you can't use env for runs-on, you also can't set default runs-on either.

Once you set a matrix somewhere, you can indeed use the variables so we can reduce, but not eliminate the copying.

**Pull request merged**